boards/arm/frdm_k22f/pinmux.c

```diff
@@ -81,6 +81,18 @@ static int frdm_k22f_pinmux_init(const struct device *dev)
 					| PORT_PCR_ODE_MASK);
 #endif
 
+#if CONFIG_SHIELD_FRDM_STBC_AGM01
```

Comment:
Is this the correct way? I mean, adding a CONFIG option for a shield in the board part itself. I remember I had a similar request adding a shield for a microphone but I seem to remember I just added the pinmux part in the shield context itself. But I'm not able to recall it neither to find this out. @erwango, do you remember?

Reply:
Indeed, when possible this should be avoided but this depends if platform has a more generic way to enable pins.
On that part I rely on @MaureenHelm

Reply:
Indeed it isn't ideal, but we don't have more generic way to configure pins on this SoC family yet.